insights/parsers/dse_ldif.py

```diff
@@ -0,0 +1,194 @@
+            kw, val = line_splits[0], line_splits[1].strip('"') if \
+                len(line_splits) == 2 else ''
+            self.lines.append(self.KeyValue(
+                kw, val, kw.lower(), line
+            ))
+        self.keywords = set([k.kw_lower for k in self.lines])
+
+    def __contains__(self, keyword):
+        """
+        (bool) Is this keyword found in the configuration file?
+        """
+        return keyword.lower() in self.keywords
+
+    def __iter__(self):
+        """
+        (iter) Yields a list of all the lines in the configuration file that
+        actually contained a configuration option.
+        """
+        for line in self.lines:
+            yield line
+
+    def __getitem__(self, keyword):
+        """
+        (list) Get all values of this keyword.
+        """
+        print('Inside __getitem__')
+        kw = keyword.lower()
+        if kw in self.keywords:
+            return [kv.value for kv in self.lines if kv.kw_lower == kw]
+
+    def get(self, keyword):
+        """
+        Get all declarations of this keyword in the configuration file.
+
+        Returns:
+            (list): a list of named tuples with the following properties:
+                - ``keyword`` - the keyword as given on that line
+                - ``value`` - the value of the keyword
+                - ``kw_lower`` - the keyword converted to lower case
+                - ``line`` - the complete line as found in the config file
+        """
+        print('Inside get()')
+        kw = keyword.lower()
+        return [kv for kv in self.lines if kv.kw_lower == kw]
+
+    def get_values(self, keyword, default='', join_with=None, split_on=None):
+        """
+        Get all the values assigned to this keyword.
+
+        Firstly, if the keyword is not found in the configuration file, the
+        value of the ``default`` option is used (defaulting to ``''``).
+
+        Then, if the ``join_with`` option is given, this string is used to
+        join the values found on each separate definition line.  Otherwise,
+        each separate definition line is returned as a string.
+
+        Finally, if the ``split_on`` option is given, this string is used to
+        split the combined string above into a list.  Otherwise, the combined
+        string is returned as is.
+        """
+        print('Inside get_values()')
+        kw = keyword.lower()
+        if kw in self.keywords:
+            vals = [kv.value for kv in self.lines if kv.kw_lower == kw]
+        else:
+            vals = [default]
+        # Should we join them together?
+        if join_with is None:
+            return vals
+        # OK, join them together.
+        val_str = join_with.join(vals)
+        # Should we now split them?
+        if split_on is None:
+            return val_str
+        else:
+            return val_str.split(split_on)
+
+    def get_line(self, keyword, default=''):
```

Comment:
The `get_line()` method is not tested. Tests are good not only for ensuring the code works as expected, but it is also an always-up-to-date form of documentation.

Reply:
@jsvob Added test for get_line